src/reducers/AuthReducer.js

```diff
@@ -1,15 +1,38 @@
+import _ from 'lodash';
+
 import {
     APP_READY,
     FACEBOOK_LOGIN_SUCCESS,
     FACEBOOK_LOGIN_FAILURE,
     FACEBOOK_LOGOUT_COMPLETE,
     GUEST_TOKEN_SET,
-    MAP_READY
+    MAP_READY,
+    FAVORITE_ADDED,
+    INITIALIZE_MAP,
+    FAVORITE_REMOVED
 } from '../actions/types';
 
 const INITIAL_STATE = {
     appLoaded: false,
-    token: null
+    token: null,
+    currentUser: {}
+};
+
+import {tokens} from '../constants';
+
+const hydrateFavorites = ({currentUser, sites}) => {
+    const preparedFavorites = _.map(currentUser.favorites, favorite => {
+        if (favorite.favoriteIsComplete) {
+            return favorite;
+        } else {
+            let matchedFavorite = _.find(sites, site => site.id === favorite.id);
+            matchedFavorite.favoriteIsComplete = true;
+
+            return matchedFavorite;
+        }
+    });
+
+    return preparedFavorites;
```

Comment:
The whole point of favorites is that they should always be available even if you have poor/no connection. Because of that, our logic here favors the local favorites data over any new data coming in. Basically, if we log in on new device, or log back in and have nothing being persisted from the phone, when we first pull down the favorite sites metadata, their `favoriteIsComplete` is `false`, since they're just text for the `id` and `title`. In that instance, we'll need to hydrate them with the full info, so we complete them. 

However, once that's done, and a user is logged in and we have a local copy of the favorite site, we flip `favoriteIsComplete` to `true`, and never look at any new info from the remote `sites` info as long as the user is still logged in. This way, even though the data may be stale, it'll *always* be available no matter what.